cluster-autoscaler/clusterstate/clusterstate_test.go

```diff
@@ -176,6 +176,73 @@ func TestHasNodeGroupStartedScaleUp(t *testing.T) {
 	}
 }
 
+func TestDeletingFailedScaleUpNodes(t *testing.T) {
```

Comment:
It's nitpicking, but I kinda don't like the name of this test. The test itself is checking that UpcomingNodes update correctly after nodeGroup resize + csr.Recalculate() which is an important assumption for the logic in core/ related to deleting failed scale-up nodes, but doesn't really have anything to do with logic related to scale-up failures in CSR. Just looking at csr it feels impossible to understand what's going on here.

Admittedly, I don't know if renaming to TestUpcomingNodesAfterResizeAndRecalculate - it's a mouthful and probably not more clear at all. Maybe what we're missing is a comment that explains what is being tested and why?

Reply:
Fair point. I've renamed the test and added a comment.